### PR TITLE
Problem panel

### PR DESCRIPTION
**Pull request by geniusgordon**

### CODE REVIEW

File: problem/models.py [hunk 1]
```diff
@@ -87,6 +87,8 @@ class Problem(models.Model):
     judge_source = models.CharField(max_length=11, choices=JUDGE_SOURCE_CHOICE, default=LOCAL)
     judge_type = models.CharField(max_length=11, choices=JUDGE_TYPE_CHOICE, default=NORMAL)
     judge_language = models.CharField(max_length=11, choices=LANGUAGE_CHOICE, default=CPP)
+    ac_count = models.IntegerField(default=0)
```
Comment (nickelcheng) [inline problem/models.py]: @henryyang42, can you calculate ac_count while migration?

Comment (henryyang42) [inline problem/models.py]: OK.

**Comment by bruce3557** [comment]
+1, wait for testing

**Comment by nickelcheng** [comment]
Problem link on nav bar should link to correct page.
If volume page doesn't used anymore, please remove it.

**Comment by nickelcheng** [comment]
Why does it still use fake data?
screenshot from 2015-04-04 23 38 06

File: problem/views.py [hunk 2]
```diff
@@ -39,9 +40,33 @@
...
+    except PageNotAnInteger:
+        all_problem = paginator.page(1)
+    except EmptyPage:
+        all_problem = paginator.page(paginator.num_pages)
+    for p in all_problem:
+        try:
+            p.pass_rate = float(p.ac_count) / float(p.total_submission) * 100.0
+            p.not_pass_rate = 100.0 - p.pass_rate
+        except ZeroDivisionError:
+            logger.info("problem %d has 0 submission" % p.pk)
```
Comment (nickelcheng) [inline problem/views.py]: What does this mean?

File: problem/views.py [hunk 3]
```diff
@@ -39,9 +40,33 @@
...
+
+    return render(request, 'problem/panel.html', 
+                  {'all_problem': all_problem, 
+                   'can_add_problem': can_add_problem})
 
 def volume(request):
```
Comment (nickelcheng) [inline problem/views.py]: if this function doesn't be used anymore, please remove it. As well as ```volume.html```

File: problem/views.py (hunk 2 above)
Comment (geniusgordon) [inline problem/views.py]: fixed

**Comment by nickelcheng** [comment]
I added a problem pid = 10463, but I can not see it in problem panel
screenshot from 2015-04-05 02 30 10

**Comment by geniusgordon** [comment]
fixed

**Comment by nickelcheng** [comment]
1. I got NameError when I want to edit problem
screenshot from 2015-04-06 22 40 31
2. Please make pass rate font size larger. It's too small to see it. (BTW, I prefer to show 2 decimal point only)
screenshot from 2015-04-06 22 42 40

File: problem/views.py [hunk 4]
```diff
@@ -86,14 +94,18 @@ def edit(request, pid=None):
         is_new = True
     if request.method == 'GET':
         if is_new:
-            form = ProblemForm()
+            form = ProblemForm(initial={'owner': request.user.username})
         else:
-            form = ProblemForm(instance=problem)
+            form = ProblemForm(instance=problem,
+                               initial={'owner': reuqest.user.username})
```
Comment (nickelcheng) [inline problem/views.py]: Is it necessary to initialize owner of a non-new problem?

File: problem/views.py [hunk 5]
```diff
@@ -86,14 +94,18 @@ def edit(request, pid=None):
...
     if request.method == 'POST':
         if is_new:
-            form = ProblemForm(request.POST)
+            form = ProblemForm(request.POST,
+                               initial={'owner': request.user.username})
         else:
-            form = ProblemForm(request.POST, instance=problem)
+            form = ProblemForm(request.POST,
+                               instance=problem,
+                               initial={'owner': reuqest.user.username})
```
Comment (nickelcheng) [inline problem/views.py]: Ditto.

**Comment by nickelcheng** [comment]
I got a NameError when I visit a nonexistent page.
screenshot from 2015-04-06 22 46 13

File: problem/views.py (hunk 4 above)
Comment (geniusgordon) [inline problem/views.py]: modified in the issue_fix branch

**Comment by geniusgordon** [comment]
fixed

**Comment by nickelcheng** [comment]
You didn't fix pass rate font size QQ

**Comment by nickelcheng** [comment]
After fix NameError issue, when a invalid page(page 0 in my testing) is requested, it just show the last page(page 230 in my testing), but URL remains invalid one. Is it correct?
screenshot from 2015-04-07 01 06 17

**Comment by nickelcheng** [comment]
+1 besides comments.
Please merge the latest code.

**Comment by geniusgordon** [comment]
if page is a integer out of range it will show the last page
if page is not an integer it will show the first page
I copied it from the django tutorial
do I need to fixed it?

**Comment by geniusgordon** [comment]
fixed

**Comment by nickelcheng** [comment]
ok, I think it's not necessary to fix page issue.

**Comment by nickelcheng** [comment]
Can you add ```No submission``` on an empty pass rate bar?

**Comment by nickelcheng** [comment]
I got ImportError, @Changron do you have any idea?
error

**Comment by geniusgordon** [comment]
added no submission

**Comment by Changron** [comment]
I'll push a hotfix

**Comment by geniusgordon** [comment]
merged latest code

File: problem/templates/problem/problem_list.html [hunk 6]
```diff
@@ -31,35 +31,55 @@
...
-              {{p.pass}}%
-            </div>
-            <div class="progress-bar progress-bar-danger" role="progressbar" aria-valuenow="{{p.not_pass}}" aria-valuemin="0" aria-valuemax="100" style="width: {{p.not_pass}}%;">
-              {{p.not_pass}}%
+          {% if p.no_submission %}
+            No submission
+          {% else %}
+            <div class="progress">
+              <div class="progress-bar progress-bar-success" style="width: {{p.pass_rate}}%;" 
+                data-toggle="tooltip" data-placement="top" title="{{p.pass_rate}}%">
```
Comment (henryyang42) [inline problem/templates/problem/problem_list.html]: Use ``floatformat`` tag may be prettier.
https://docs.djangoproject.com/en/1.8/ref/templates/builtins/#floatformat

**Comment by nickelcheng** [comment]
Why does pass rate bar looks different when I use different account to login?
ugly

**Comment by henryyang42** [comment]
The theme you choose is different

**Comment by nickelcheng** [comment]
OK. I suppose every theme have same progress bar. Is that possible?

**Comment by nickelcheng** [comment]
+1, please merge the latest code

**Comment by nickelcheng** [comment]
going to merge

**Pull request merged**